### PR TITLE
Pagination helpers

### PR DESCRIPTION
**Pull request by aenario**
This PR includes 2 changes related to pagination

- For mango requests results, we add some documentation and a `next` and `limit` to requests answer 
- For /data/type/id/references, we use jsonapi pagination with cursor

### CODE REVIEW

File: docs/mango.md [hunk 1]
```diff
@@ -116,3 +116,49 @@ Content-Type: application/json
...
+
+```json
+{
+    "limit": 100,
+    "next": true,
+    "docs": [ "... first hundred docs ..."],
+}
+```
+
+If there is the number of docs is lower or equal to the limit, these fields are not present
```
Comment (jinroh) [inline docs/mango.md]: `s/there is/the/` ?

File: pkg/couchdb/cursor.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		req.Limit = c.Limit + 1
+	}
+
+	return req
+}
+
+// GetNextCursor returns a cursor to the end of a ViewResponse
+// it removes the last item from the response to create a Cursor
+func GetNextCursor(res *ViewResponse) *Cursor {
+	next := res.Rows[len(res.Rows)-1]
```
Comment (jinroh) [inline pkg/couchdb/cursor.go]: shouldn't we have a check when `res.Rows` is of length 0 ?

File: web/data/data.go [hunk 3]
```diff
@@ -295,13 +297,28 @@ func findDocuments(c echo.Context) error {
 		return err
 	}
 
+	limit, hasLimit := findRequest["limit"].(float64)
+	if !hasLimit || limit > maxMangoLimit {
+		limit = 100
+	}
+
+	// add 1 so we know if there is more.
+	findRequest["limit"] = limit + 1
```
Comment (jinroh) [inline web/data/data.go]: couchdb does not provide a better way than that ?

File: web/data/data.go [hunk 4]
```diff
@@ -295,13 +297,28 @@ func findDocuments(c echo.Context) error {
 		return err
 	}
 
+	limit, hasLimit := findRequest["limit"].(float64)
```
Comment (jinroh) [inline web/data/data.go]: why `float64` and not `int` ?

File: docs/mango.md [hunk 5]
```diff
@@ -116,3 +116,49 @@ Content-Type: application/json
...
+
+```json
+{
+    "docs": [ "... less than a hundred docs ..."],
+}
+```
```
Comment (jinroh) [inline docs/mango.md]: Maybe I'm wrong, but I would better see an api where we always put the `next` and `limit` field (even if they are `false` and the default value) to be more explicit ?

Also IMHO, it is better than having to handle `undefined`s.

File: docs/mango.md [hunk 6]
```diff
@@ -116,3 +116,49 @@ Content-Type: application/json
...
+Pagination of mango query should be handled by the client.
+The stack will always limit query results to maximum 100 docs.
+
+If the limit cause some docs to not be returned, the response will have a
+limit & next values.
+
+```json
+{
+    "limit": 100,
+    "next": true,
```
Comment (nono) [inline docs/mango.md]: In `next`, I'd have put the URL for asking the next documents (if it is not too complicated to compute)

File: web/data/data.go (hunk 4 above)
Comment (nono) [inline web/data/data.go]: Go use `float64` when deserializing a JSON to a `map[string]interface{}`

File: web/data/data.go (hunk 3 above)
Comment (aenario) [inline web/data/data.go]: nope http://docs.couchdb.org/en/2.0.0/couchapp/views/pagination.html#paging-alternate-method

File: docs/mango.md (hunk 5 above)
Comment (aenario) [inline docs/mango.md]: good idea, will do.

File: docs/mango.md (hunk 6 above)
Comment (aenario) [inline docs/mango.md]: It is a bit complicated to compute : we would need to have the index definition to see which field we can/should use as a cursor but the /data/xxxx/_find route does not have it.

Comment (nono) [inline docs/mango.md]: OK, let's forgot it

**Pull request merged**